tests/integration/test_metadata_model.py

```diff
@@ -54,31 +143,70 @@ def test_submit_filebased_manifest(
         # GIVEN a metadata model object using class labels
         meta_data_model = metadata_model(helpers, "class_label")
 
-        # AND a filebased test manifset
-        manifest_path = helpers.get_data_path(manifest_path)
+        # AND a filebased test manifest
+        load_args = {"dtype": "string"}
+        manifest = helpers.get_data_frame(
+            manifest_path, preserve_raw_input=True, **load_args
+        )
+        manifest_full_path = helpers.get_data_path(manifest_path)
 
-        # WHEN the manifest it submitted
+        # WHEN the manifest is submitted and files are annotated
         # THEN submission should complete without error
         with does_not_raise():
             manifest_id = meta_data_model.submit_metadata_manifest(
-                manifest_path=manifest_path,
+                manifest_path=manifest_full_path,
                 dataset_id=dataset_id,
-                manifest_record_type="file_and_entities",
+                manifest_record_type=manifest_record_type,
                 restrict_rules=False,
                 file_annotations_upload=True,
                 hide_blanks=False,
                 validate_component=validate_component,
                 dataset_scope=dataset_scope,
             )
 
-            # AND the manifest should be submitted to the correct place
-            assert manifest_id == expected_manifest_id
+        # AND the files should be annotated
+        spy_add_annotations.assert_called_once()
 
-            # AND the manifest should be uploaded as a CSV
-            spy_upload_file_as_csv.assert_called_once()
-            # AND annotations should be added to the files
-            spy_add_annotations.assert_called_once()
+        # AND the annotations on the entities should have the correct metadata
+        for index, row in manifest.iterrows():
+            entityId = row["entityId"]
+            expected_sample_id = row["Sample ID"]
+            annos = synapse_store.syn.get_annotations(entityId)
+            sample_id = annos["SampleID"][0]
+            assert str(sample_id) == str(expected_sample_id)
+
+        # AND the annotations on the manifest file itself should have the correct metadata
+        manifest_annos = synapse_store.syn.get_annotations(manifest_id)
+        for annotation in manifest_annos.keys():
```

Comment:
Hi @linglp - sounds good! To clarify before I move forward: I generalized this logic to validate the annotations for the manifest file/table because [this docstring ](https://github.com/Sage-Bionetworks/schematic/blob/5721c77d35ce14b45c7bf768af133c6d81eb4abe/schematic/store/synapse.py#L1606) was implying that, in the future, the manifest files/tables will be annotated with more annotations beyond `Component`.

With that in mind, I'll be keeping that generalized logic so that developers don't have to change this test in the future, but I will patch the other checks for `eTag` and `entityId` so it's easier to read. Thanks!

Reply:
Oh I see. Thanks for pointing that out. What you proposed makes sense. Thank you!

Reply:
[Updated](https://github.com/Sage-Bionetworks/schematic/pull/1485/commits/e3da3d2cf25a6e31e1a17be2d91878d01aa70264)